api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TXCurrQueries.java

```diff
@@ -0,0 +1,309 @@
+            + "and pg.program_id=%s and  ps.state=%s and pg.date_enrolled <= :onOrBefore  "
+            + "and pg.location_id= :location group by p.patient_id ";
+
+    return String.format(query, aRTProgram, artTransferredFromOtherHealthFacilityWorkflowState);
+  }
+
+  public static String getPatientWithFirstDrugPickupEncounterBeforeOrOnEndDate(
+      int aRVPharmaciaEncounterType) {
+
+    String query =
+        "SELECT p.patient_id "
+            + "FROM patient p "
+            + "INNER JOIN encounter e ON p.patient_id=e.patient_id "
+            + "WHERE p.voided=0 AND e.encounter_type=%d "
+            + "AND e.voided=0 AND e.encounter_datetime <= :onOrBefore AND e.location_id=:location GROUP BY p.patient_id";
+
+    return String.format(query, aRVPharmaciaEncounterType);
+  }
+
+  public static String getPatientsDeadTransferredOutSuspensionsInProgramStateByReportingEndDate(
+      int artProgram,
+      int transferredOutToAnotherHealthFacilityWorkflowState,
+      int suspendedTreatmentWorkflowState,
+      int artDeadWorkflowState) {
+
+    String query =
+        " select p.patient_id from patient p "
+            + " inner join patient_program pg on p.patient_id=pg.patient_id "
+            + " inner join patient_state ps on pg.patient_program_id=ps.patient_program_id "
+            + " where pg.voided=0 and ps.voided=0 and p.voided=0 and pg.program_id=%d "
+            + " and ps.state in (%d,%d,%d) and ps.end_date is null and ps.start_date<=:onOrBefore "
+            + "and pg.location_id=:location group by p.patient_id  ";
+
+    return String.format(
+        query,
+        artProgram,
+        transferredOutToAnotherHealthFacilityWorkflowState,
+        suspendedTreatmentWorkflowState,
+        artDeadWorkflowState);
+  }
+
+  public static String getDeadPatientsInFichaResumeAndClinicaOfMasterCardByReportEndDate() {
+
+    String query =
+        "select p.patient_id  "
+            + " from patient p "
+            + " inner join person prs on prs.person_id=p.patient_id "
+            + " inner join  encounter e on  e.patient_id=p.patient_id "
+            + " where prs.dead=1 and prs.death_date <= :onOrBefore and p.voided=0 and prs.voided=0 "
+            + " and e.location_id = :location group by p.patient_id ";
+
+    return query;
+  }
+
+  public static String getPatientDeathRegisteredInLastHomeVisitCardByReportingEndDate(
+      String encounterTypes,
+      int patientFoundConcept,
+      int reasonPatientNotFound,
+      int noConcept,
+      int patientIsDead) {
+    String query =
+        "select  p.patient_id"
+            + " from patient p "
+            + " inner join encounter e on e.patient_id=p.patient_id "
+            + " inner join obs o  on o.encounter_id=e.encounter_id "
+            + " where  p.voided=0  and e.voided=0 and o.voided=0 "
+            + " and e.encounter_type in (%s) and o.concept_id in (%s,%s) and o.value_coded in (%s,%s) "
```

Comment:
So this is checking that the patient has any concept matched with any value_coded, but the specs are actually more specific than that:
```Patient not found (Concept ID 2003) = NO (Concept ID 1066)
Reason of Not Finding (Concept ID 2031) = Died (COncpet Id 1383)```
So we should actually be pairing the concept with the value coded value - to be match the specs closer

Reply:
I  think here  I missed the last  encounter  as the spec  says....